From 4c949d3f2f6f2e13cd0b42477bb28a7f1601735a Mon Sep 17 00:00:00 2001
From: nx111 <NX111.AimH@gmail.com>
Date: Wed, 2 Jan 2019 20:24:54 +0800
Subject: [PATCH 1/2] Messaging: Implement option for notification quick reply.

Showing quick reply items in the notification bar is too messy,
add an option to switch it on or off. default to off.

Change-Id: I15fdbe5d494a764d094447a19972a9fe2269232b
---
 res/values/cm_strings.xml                              |  3 +++
 res/values/lineage_constants.xml                       |  3 +++
 res/xml-v21/preferences_application.xml                |  7 +++++++
 res/xml-v23/preferences_application.xml                |  7 +++++++
 res/xml/preferences_application.xml                    |  7 +++++++
 .../messaging/datamodel/BugleNotifications.java        |  6 +++++-
 src/org/lineageos/messaging/util/PrefsUtils.java       | 10 +++++++++-
 7 files changed, 41 insertions(+), 2 deletions(-)

diff --git a/res/values/cm_strings.xml b/res/values/cm_strings.xml
index 1c902a7..1203313 100644
--- a/res/values/cm_strings.xml
+++ b/res/values/cm_strings.xml
@@ -38,4 +38,7 @@
     <!-- Notification channel -->
     <string name="notification_channel_title">Messages</string>
     <string name="background_worker_notif">Checking messages\u2026</string>
+
+    <!-- Notification Reply -->
+    <string name="notification_reply_pref">Quick reply notification</string>
 </resources>
diff --git a/res/values/lineage_constants.xml b/res/values/lineage_constants.xml
index cf2793b..6218e1d 100644
--- a/res/values/lineage_constants.xml
+++ b/res/values/lineage_constants.xml
@@ -25,4 +25,7 @@
     <!-- Preference keys for user-visible settings -->
     <!-- Application-wide settings -->
     <bool name="show_emoticons_pref_default" translatable="false">true</bool>
+
+    <!-- Notification Reply -->
+    <bool name="notification_reply_pref_default">false</bool>
 </resources>
diff --git a/res/xml-v21/preferences_application.xml b/res/xml-v21/preferences_application.xml
index c403c06..e2434ef 100644
--- a/res/xml-v21/preferences_application.xml
+++ b/res/xml-v21/preferences_application.xml
@@ -66,6 +66,13 @@
         android:persistent="true"
         android:dependency="@string/notifications_enabled_pref_key" />
 
+    <SwitchPreference
+        android:key="pref_notification_reply"
+        android:title="@string/notification_reply_pref"
+        android:defaultValue="@bool/notification_reply_pref_default"
+        android:persistent="true"
+        android:dependency="@string/notifications_enabled_pref_key" />
+
     <SwitchPreference
         android:key="pref_show_emoticons"
         android:title="@string/show_emoticons_pref_title"
diff --git a/res/xml-v23/preferences_application.xml b/res/xml-v23/preferences_application.xml
index d9cc151..10be11f 100644
--- a/res/xml-v23/preferences_application.xml
+++ b/res/xml-v23/preferences_application.xml
@@ -67,6 +67,13 @@
         android:persistent="true"
         android:dependency="@string/notifications_enabled_pref_key" />
 
+    <SwitchPreference
+        android:key="pref_notification_reply"
+        android:title="@string/notification_reply_pref"
+        android:defaultValue="@bool/notification_reply_pref_default"
+        android:persistent="true"
+        android:dependency="@string/notifications_enabled_pref_key" />
+
     <SwitchPreference
         android:key="pref_show_emoticons"
         android:title="@string/show_emoticons_pref_title"
diff --git a/res/xml/preferences_application.xml b/res/xml/preferences_application.xml
index 03b868b..02e8011 100644
--- a/res/xml/preferences_application.xml
+++ b/res/xml/preferences_application.xml
@@ -66,6 +66,13 @@
         android:persistent="true"
         android:dependency="@string/notifications_enabled_pref_key" />
 
+    <SwitchPreference
+        android:key="pref_notification_reply"
+        android:title="@string/notification_reply_pref"
+        android:defaultValue="@bool/notification_reply_pref_default"
+        android:persistent="true"
+        android:dependency="@string/notifications_enabled_pref_key" />
+
     <CheckBoxPreference
         android:key="pref_show_emoticons"
         android:title="@string/show_emoticons_pref_title"
diff --git a/src/com/android/messaging/datamodel/BugleNotifications.java b/src/com/android/messaging/datamodel/BugleNotifications.java
index 2127d81..ddd6c6f 100644
--- a/src/com/android/messaging/datamodel/BugleNotifications.java
+++ b/src/com/android/messaging/datamodel/BugleNotifications.java
@@ -89,6 +89,8 @@ import java.util.List;
 import java.util.Locale;
 import java.util.Set;
 
+import org.lineageos.messaging.util.PrefsUtils;
+
 /**
  * Handle posting, updating and removing all conversation notifications.
  *
@@ -926,7 +928,9 @@ public class BugleNotifications {
                 setChoices(choices)
                 .build();
         actionBuilder.addRemoteInput(remoteInput);
-        notifBuilder.addAction(actionBuilder.build());
+        if (PrefsUtils.isNotificationReplyEnabled()) {
+            notifBuilder.addAction(actionBuilder.build());
+        }
 
         // Support the action on a wearable device as well
         wearableExtender.addAction(actionBuilder.build());
diff --git a/src/org/lineageos/messaging/util/PrefsUtils.java b/src/org/lineageos/messaging/util/PrefsUtils.java
index 06c61ed..d6286bb 100644
--- a/src/org/lineageos/messaging/util/PrefsUtils.java
+++ b/src/org/lineageos/messaging/util/PrefsUtils.java
@@ -23,7 +23,7 @@ import com.android.messaging.util.BuglePrefs;
 
 public class PrefsUtils {
     public static final String SHOW_EMOTICONS_ENABLED    = "pref_show_emoticons";
-
+    public static final String NOTIFICATION_REPLY_ENABLED = "pref_notification_reply";
     private PrefsUtils() {
         //Don't instantiate
     }
@@ -49,4 +49,12 @@ public class PrefsUtils {
                 R.bool.show_emoticons_pref_default);
         return prefs.getBoolean(SHOW_EMOTICONS_ENABLED, defaultValue);
     }
+
+    public static boolean isNotificationReplyEnabled() {
+        final BuglePrefs prefs = BuglePrefs.getApplicationPrefs();
+        final Context context = Factory.get().getApplicationContext();
+        final boolean defaultValue = context.getResources().getBoolean(
+                R.bool.notification_reply_pref_default);
+        return prefs.getBoolean(NOTIFICATION_REPLY_ENABLED, defaultValue);
+    }
 }
-- 
2.17.1

